dbms/src/DataStreams/ParallelInputsProcessor.h

```diff
@@ -106,6 +106,7 @@ class ParallelInputsProcessor
         , additional_input_at_end(additional_input_at_end_)
         , max_threads(std::min(inputs_.size(), max_threads_))
         , handler(handler_)
+        , may_available_inputs(inputs_.size())
```

Comment:
I think we should update this vairable in L232, it looks a little unsafe to me to set the value directly here.